mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -334,7 +334,15 @@ class AffineYieldOpLowering : public OpRewritePattern<AffineYieldOp> {
 
   LogicalResult matchAndRewrite(AffineYieldOp op,
                                 PatternRewriter &rewriter) const override {
-    rewriter.replaceOpWithNewOp<scf::YieldOp>(op);
+    if (isa<scf::ParallelOp>(op.getParentOp())) {
+      // This could happen while lowering affine.parallel to scf.parallel.
+      // scf.parallel in all cases expect scf.yield to have no operands.
+      rewriter.replaceOpWithNewOp<scf::YieldOp>(op);
+      return success();
+    }
+    SmallVector<Value, 4> yieldOperands(op.operands().begin(),
+                                        op.operands().end());
+    rewriter.replaceOpWithNewOp<scf::YieldOp>(op, yieldOperands);
```

Comment:
You don't need to materialize the operands into a small vector. You can directly forward them and the temporary allocation.

Reply:
Done.